projects/plugins/jetpack-mu-wpcom/src/features/coming-soon/coming-soon.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * Coming Soon
+ *
+ * @package A8C\FSE\Coming_soon
+ */
+
+namespace A8C\FSE\Coming_soon;
+
+/**
+ * Determines whether the coming soon page should be shown.
+ *
+ * @return boolean
+ */
+function should_show_coming_soon_page() {
+	if ( ! is_singular() && ! is_archive() && ! is_search() && ! is_front_page() && ! is_home() && ! is_404() ) {
+		return false;
+	}
+
+	$share_code = get_share_code();
+	if ( is_accessed_by_valid_share_link( $share_code ) ) {
+		track_preview_link_event();
+		setcookie( 'wp_share_code', $share_code, time() + 3600, '/', false, is_ssl(), true );
+		return false;
+	}
+
+	$should_show = ( (int) get_option( 'wpcom_public_coming_soon' ) === 1 );
+
+	// Everyone from Administrator to Subscriber will be able to see the site.
+	// See https://wordpress.org/support/article/roles-and-capabilities/ for all roles and capabilities.
+	// We can update to `edit_post` to be stricter, or open it up as an editable feature.
+	if ( is_user_logged_in() && current_user_can( 'read' ) ) {
+		$should_show = false;
+	}
+
+	// Allow folks to hook into this method to set their own rules.
+	// We'll use to on WordPress.com to check further user privileges.
+	return apply_filters( 'a8c_show_coming_soon_page', $should_show );
+}
+
+/**
+ * Gets share code from URL or Cookie.
+ *
+ * @return string
+ */
+function get_share_code() {
+	if ( isset( $_GET['share'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Might be nice to improve these comments while we're here. p9dueE-4z8-p2

I'm guessing this 'share' parameter doesn't write anything to the site, it just lets someone see the page without "coming soon" getting in the way.

Reply:
33ee54fc154f2a7c82b4bc10a246c70ecaf3b650